client/packages/common/src/plugins/hooks/usePluginColumns.ts

```diff
@@ -3,21 +3,22 @@ import { ColumnPluginType } from '../types';
 import { RecordWithId } from '../../types/utility';
 import { ColumnDefinition } from '../../ui';
 import { usePluginProvider } from '../components';
+import { ArrayUtils } from '../../utils/arrays';
 
 export function usePluginColumns<T extends RecordWithId>({
   type,
 }: {
   type: ColumnPluginType;
 }) {
-  const { columnPlugins } = usePluginProvider();
+  const columnPlugins = usePluginProvider(state => state.columnPlugins);
   const [pluginColumns, setPluginColumns] = useState<ColumnDefinition<T>[]>([]);
   const columns = columnPlugins.filter(column => column.type === type);
 
   useEffect(() => {
     columns.forEach(plugin => {
       plugin.column().then(column => {
         const mapped = column as unknown as ColumnDefinition<T>;
-        setPluginColumns([...pluginColumns, mapped]);
+        setPluginColumns(ArrayUtils.uniqBy([...pluginColumns, mapped], 'key'));
```

Comment:
if the same column is added again, we only want one of them! I've seen this with HMR, but it's possible it will happen under normal conditions